src/builtins.ts

```diff
@@ -771,6 +772,44 @@ export function compileCall(
         }
       }
     }
+    case BuiltinSymbols.nameof: {
+      let value: string | null = null;
+      let resultType: Type | null = evaluateConstantType(compiler, typeArguments, operands, reportNode);
+      compiler.currentType = compiler.program.stringInstance.type;
+      if (resultType === null) {
+        compiler.error(
+          DiagnosticCode.Operation_not_supported,
+          reportNode.typeArgumentsRange
+        );
+        return module.unreachable();
+      }
+
+      if (resultType.classReference !== null) {
+        value = resultType.classReference.name;
+      } else if (resultType.signatureReference !== null) {
+        value = "Function";
+      } else {
+        switch (resultType.kind) {
+          case TypeKind.BOOL: value = "bool"; break;
+          case TypeKind.I8: value = "i8"; break;
+          case TypeKind.U8: value = "u8"; break;
+          case TypeKind.I16: value = "i16"; break;
+          case TypeKind.U16: value = "u16"; break;
+          case TypeKind.I32: value = "i32"; break;
+          case TypeKind.U32: value = "u32"; break;
+          case TypeKind.F32: value = "f32"; break;
+          case TypeKind.I64: value = "i64"; break;
+          case TypeKind.U64: value = "u64"; break;
+          case TypeKind.F64: value = "f64"; break;
+          case TypeKind.ISIZE: value = "isize"; break;
+          case TypeKind.USIZE: value = "usize"; break;
+          case TypeKind.VOID: value = "void"; break;
+          default: value = "unknown"; break;
```

Comment:
Missing `v128` to cover all possible cases. Once added, the `default` case can be moved to the top with only `assert(false)` in it.

Reply:
I added `v128` but I don't understand why we should `assert(false)` in the invalid case. I'm happy to comply with the request, however. Would you mind explaining why?

Reply:
Since `TypeKind` is an enum and the switch here is expected to handle all possible values, any other value than declared in the enum would be an error. Also, moving the default case above the void case will make sure that `value` can't be uninitialized.

Reply:
Got it. I think setting `value = ""` might be a cleaner solution, but I went with your suggestion instead. I'll let you decide which one you like better.

Reply:
Just want to make sure this is the way you want to go before I mark this as resolved.

Reply:
Yes, it's fine now :)